include/aws/common/bigint.h

```diff
@@ -0,0 +1,144 @@
+ * calling aws_bigint_init_from_hex() to generate an arbitrary negative number.
+ */
+AWS_COMMON_API
+int aws_bigint_init_from_hex(
+    struct aws_bigint *bigint,
+    struct aws_allocator *allocator,
+    struct aws_byte_cursor hex_digits);
+
+/**
+ * Initializes a big int from a 64 bit signed integer
+ */
+AWS_COMMON_API
+int aws_bigint_init_from_int64(struct aws_bigint *bigint, struct aws_allocator *allocator, int64_t value);
+
+/**
+ * Initializes a big int from a 64 bit unsigned integer
+ */
+AWS_COMMON_API
+int aws_bigint_init_from_uint64(struct aws_bigint *bigint, struct aws_allocator *allocator, uint64_t value);
+
+/**
+ * Initializes a big int as a copy of another big int
+ */
+AWS_COMMON_API
+int aws_bigint_init_from_copy(struct aws_bigint *bigint, const struct aws_bigint *source);
+
+/**
+ * Writes a bigint to a buffer as a hexadecimal number.  Will prepend (-) in front of negative numbers for
+ * easier testing.  This API is primarily intended for testing.  Actual output (to various formats/bases) is TBD.
+ */
+AWS_COMMON_API
+int aws_bigint_bytebuf_debug_output(const struct aws_bigint *bigint, struct aws_byte_buf *buffer);
+
+/**
+ * Returns true if this integer is negative, false otherwise.
+ */
+AWS_COMMON_API
+bool aws_bigint_is_negative(const struct aws_bigint *bigint);
+
+/**
+ * Returns true if this integer is positive, false otherwise.
+ */
+AWS_COMMON_API
+bool aws_bigint_is_positive(const struct aws_bigint *bigint);
+
+/**
+ * Returns true if this integer is zero, false otherwise.
+ */
+AWS_COMMON_API
+bool aws_bigint_is_zero(const struct aws_bigint *bigint);
+
+/**
+ * Returns true if the two big ints are equal in value, false otherwise
+ */
+AWS_COMMON_API
+bool aws_bigint_equals(const struct aws_bigint *lhs, const struct aws_bigint *rhs);
+
+/**
+ * Returns true if the two big ints are not equal in value, false otherwise
+ */
+AWS_COMMON_API
+bool aws_bigint_not_equals(const struct aws_bigint *lhs, const struct aws_bigint *rhs);
+
+/**
+ * Returns true if the first operand is less than the second operand
+ */
+AWS_COMMON_API
+bool aws_bigint_less_than(const struct aws_bigint *lhs, const struct aws_bigint *rhs);
+
+/**
+ * Returns true if the first operand is less than or equal to the second operand
+ */
+AWS_COMMON_API
+bool aws_bigint_less_than_or_equals(const struct aws_bigint *lhs, const struct aws_bigint *rhs);
+
+/**
+ * Returns true if the first operand is greater than the second operand
+ */
+AWS_COMMON_API
+bool aws_bigint_greater_than(const struct aws_bigint *lhs, const struct aws_bigint *rhs);
+
+/**
+ * Returns true if the first operand is greater than or equal to the second operand
+ */
+AWS_COMMON_API
+bool aws_bigint_greater_than_or_equals(const struct aws_bigint *lhs, const struct aws_bigint *rhs);
+
+/**
+ * Negates the supplied bigint.  Has no effect on a zero-valued integer.
+ */
+AWS_COMMON_API
+void aws_bigint_negate(struct aws_bigint *bigint);
+
+AWS_EXTERN_C_END
```

Comment:
no arithmetic operations?

Reply:
Separate reviews.  For example https://github.com/awslabs/aws-c-common/pull/588